api/v1beta1/imagerepository_types.go

```diff
@@ -67,6 +67,19 @@ type ImageRepositorySpec struct {
 	// It does not apply to already started scans. Defaults to false.
 	// +optional
 	Suspend bool `json:"suspend,omitempty"`
+
+	// AccessFrom defines an ACL for allowing cross-namespace references
+	// to the ImageRepository object based on the caller's namespace labels.
+	// +optional
+	AccessFrom *AccessFrom `json:"accessFrom,omitempty"`
+}
+
+type AccessFrom struct {
+	NamespaceSelectors []NamespaceSelector `json:"namespaceSelector,omitempty"`
```

Comment:
The field name and the JSON name don't match. I think the plural is probably what you want:

```yaml
accessFrom:
  namespaceSelectors:
  - matchLabels: {k8s.io/namespace: podinfo-app} # or whatever the official label is
  - matchLabels: {use: app, app: podinfo} # just to make some up ...
```

Reply:
The naming is the same as with network policies `namespaceSelector`.

Reply:
Yes, but there it is a matchLabels, not a list of matchLabels. The factoring for network policy is

```yaml
  from:
    - ipBlock:
        cidr: 172.17.0.0/16
        except:
        - 172.17.1.0/24
    - namespaceSelector:
        matchLabels:
          project: myproject
    - podSelector:
        matchLabels:
          role: frontend
```

as in, there's a list of different ways of selecting pods, one kind of which is `NamespaceSelector`, which has a single `matchLabels`.

Reply:
If you want it to be the same as network policy, you should have:

```yaml
allowFrom:
- namespaceSelector:
    matchLabels: {...}
- namespaceSelector:
    matchLabels: {...}
```

Reply:
Fixed!